### PR TITLE
docs: document the x-deprecated option

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

The `x-deprecated` option isn't documented AFAIK.
I think this can be helpful to mention it for library authors.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/schematics-authoring.md [hunk 1]
```diff
@@ -91,6 +91,29 @@ For example, the hypothetical "Hello World" schematic might have the following s
 
 You can see examples of schema files for the Angular CLI command schematics in [`@schematics/angular`](https://github.com/angular/angular-cli/blob/7.0.x/packages/schematics/angular/application/schema.json).
 
+
+#### Deprecated options
+
+To flag an option as deprecated, you can use the `x-deprecated` property.
+The assigned value will be used to warn the user when the deprecated option is used.
```
Comment (kapunahelewong) [inline aio/content/guide/schematics-authoring.md:98]:
```suggestion
Angular uses the assigned value when notifying the user that an option is deprecated.
```

Or something like this that removes the future tense and the passive. Is it still correct with "Angular" as the subject? I could be wrong on that one. If the subject could be more specific than Angular, that'd be even better.

**Review by kapunahelewong — COMMENTED** [review]
Just one small suggestion. This will still need engineering review. Thank you, @timdeschryver!

File: aio/content/guide/schematics-authoring.md (hunk 1 above)
Comment (timdeschryver) [reply aio/content/guide/schematics-authoring.md]: Thanks @kapunahelewong 
I'm currently taking the technical writer course and this is a great example of it 💯

**Comment by petebacondarwin** [comment]
There is some work post-v12 to migrate to a more recent version of json-schema. And we don't want to document this custom property until we have stabilized that. So I am going to mark this PR as blocked for now.

**Comment by dylhunn** [comment]
I spoke to Alan, and he gave me the following info:

> I don't think we should put it yet in the docs. I think we need to define the api of schematics a bit better before doing so. If anything also, we should document all the supported schema properties and not x-deprecated.

I will close this for now, and we can revisit it in the future if the above is achieved.

**Pull request closed without merging**